packages/components/src/components/amount-input/amount-input.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useCallback, useState } from 'react';
+import { isMobile } from '@deriv/shared';
+import Input from '../input';
+import Text from '../text';
+
+type TAmountInput = {
+    currency: string;
+    decimal_places?: number;
+    disabled?: boolean;
+    initial_value?: number;
+    label?: string;
+    max_digits?: number;
+    onChange?: (value: number) => void;
+};
+
+const AmountInput = ({
+    currency,
+    decimal_places = 2,
+    disabled = false,
+    initial_value = 0,
+    label,
+    max_digits = 8,
+    onChange,
+}: TAmountInput) => {
+    const [value, setValue] = useState(initial_value);
+    const [focus, setFocus] = useState(false);
+
+    const displayNumber = useCallback(
+        (number: number) => number.toLocaleString(undefined, { minimumFractionDigits: decimal_places }),
+        [decimal_places]
+    );
+
+    const onChangeHandler = (e: { target: { value: string } }) => {
```

Comment:
```suggestion
    const onChangeHandler: React.ComponentProps<typeof Input>['onChange'] = e => {
```

Reply:
Been discussing typing here with @heorhi-deriv, as the `React.ChangeEvent<HTMLInputElement>` doesn't work here, so I went with this duck typing
The way you suggest is much better, thanks